hydrocronapi/data_access/db.py

```diff
@@ -18,101 +20,139 @@
 
 
 def get_reach_series(start_time: datetime, end_time: datetime) -> Generator:
-    try:
-        conn = pymysql.connect(
-            host=DB_HOST, user=DB_USERNAME, passwd=DB_PASSWORD, database=DB_NAME, connect_timeout=10, charset='utf8mb4',
-            cursorclass=pymysql.cursors.DictCursor
-        )
-    except pymysql.MySQLError as e:
-        logger.error("ERROR: Unexpected error: Could not connect to MySql instance.")
-        logger.error(e)
-        sys.exit()
-
-    logger.info("SUCCESS: Connection to mysql instance succeeded")
-
-    st = float(time.mktime(start_time.timetuple()) - 946710000)
-    et = float(time.mktime(end_time.timetuple()) - 946710000)
-
-    with conn.cursor() as cur:
-        cur.execute(
-            """select * from reach where cast(time as float) >= %(start_time)s and cast(time as float) <= %(end_time)s""",
-            {'start_time': st, 'end_time': et}
-        )
-        for row in cur:
-            yield row
+
+    session = boto3.session.Session(aws_access_key_id='a',
```

Comment:
Where and how do we do session management for the database connection? Are there limits to concurrent open sessions to dynamo? Performance implications? Right now this is opening/closing a new session per-request which may not be an efficient way of doing this

Reply:
I'm not an expert in DynamoDB but I believe concurrent read-only access can be configured directly in AWS without having to change the code.
Do you have any alternative suggestion or example on how to handle these sessions?

Reply:
New code in PR #issue-14 that might address this issue, as it doesn't implement boto3 sessions.